### PR TITLE
Adding better handling of broker service exceptions and additional logging

### PR DESCRIPTION
**Pull request by trwalke**
fix for https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1938

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs [hunk 1]
```diff
@@ -83,7 +83,12 @@ public async Task<MsalTokenResponse> FetchTokensAsync(CancellationToken cancella
             {
                 _logger.Info(
                     LogMessages.ErrorReturnedInBrokerResponse(msalTokenResponse.Error));
-                throw new MsalServiceException(msalTokenResponse.Error, MsalErrorMessage.BrokerResponseError + msalTokenResponse.ErrorDescription);
+
+                throw MsalServiceExceptionFactory.FromBrokerResponse(msalTokenResponse.Error,
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Internal/Broker/BrokerInteractiveRequestComponent.cs:87]: Do we always get Error, SubError etc. ? If not, we should add some sensible defaults, smth like "unkown_broker_error" "Something happened. Please collect broker logs". 

Currently if we don't know the error details we throw smth related to AndroidBrokerCannotBeInvoked, which is very misleading.

File: src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBroker.cs [hunk 2]
```diff
@@ -249,7 +259,7 @@ public async Task<IEnumerable<IAccount>> GetAccountsAsync(string clientID, strin
             {
                 if (!IsBrokerInstalledAndInvokable())
                 {
-                    _logger.Error("Android broker is not installed so no accounts will be returned.");
+                    _logger.Error("Android broker is either not installed or is not reachable so no accounts will be returned.");
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBroker.cs:262]: Can you lower the level for this statement please from Error to Info or Warn?

File: src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBroker.cs [hunk 3]
```diff
@@ -276,7 +286,7 @@ public async Task RemoveAccountAsync(string clientID, IAccount account)
             {
                 if (!IsBrokerInstalledAndInvokable())
                 {
-                    _logger.Error("Android broker is not installed so no accounts will be removed.");
+                    _logger.Error("Android broker is either not installed or not reachable so no accounts will be removed.");
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBroker.cs:289]: Same here.

File: tests/Microsoft.Identity.Test.Unit.net45/RequestsTests/BrokerRequestTests.cs [hunk 4]
```diff
@@ -81,6 +81,50 @@ public void BrokerErrorResponseTest()
...
+        {
+            CreateBrokerHelper();
+
+            var response = new MsalTokenResponse
+            {
+                Error = MsalError.InvalidGrantError,
+                ErrorDescription = MsalError.InvalidGrantError
+            };
+
+            ValidateBrokerResponse(
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Unit.net45/RequestsTests/BrokerRequestTests.cs:117]: Please update validation code to check for UiRequiredException and for other exception details like the http stuff.

Comment (trwalke) [reply tests/Microsoft.Identity.Test.Unit.net45/RequestsTests/BrokerRequestTests.cs:117]: This already checks for uirequiredexceptions via the callback. adding other http stuff

**Pull request merged**